### PR TITLE
Faster TXGen with StateSync

### PR DESCRIPTION
**Pull request by alajko**
## Issue

Inorder to be able to generate txns, we need the txn to be able to sync to current status of the blockchain.

### CODE REVIEW

**Comment by rlan35** [comment]
This is separating txgen all-in-one model to one per shard model? This will make cross-shard txs hard. Why do you have to change the model?

**Comment by LeoHChen** [comment]
please squash your commit into fewer commits, and the commit messages have to be meaningful.

RJ also mentioned don't use isBeacon anymore.  You may need to find another way of doing this.
Also, Jenkins failed.  Please make sure it passed the Jenkins test.

File: node/node_syncing.go [hunk 1]
```diff
@@ -41,6 +41,7 @@ func (node *Node) getNeighborPeers(neighbor *sync.Map) []p2p.Peer {
 
 // GetBeaconSyncingPeers returns a list of peers for beaconchain syncing
 func (node *Node) GetBeaconSyncingPeers() []p2p.Peer {
+	utils.GetLogInstance().Info("[SYNC] BeaconSyncing Peers is called", "node", node.SelfPeer)
```
Comment (LeoHChen) [inline node/node_syncing.go]: no need to add debug message here.

File: node/node_syncing.go [hunk 2]
```diff
@@ -109,6 +110,12 @@ func (node *Node) SupportSyncing() {
 	}
 }
 
+// GetSync gets sync.
+func (node *Node) GetSync() {
```
Comment (LeoHChen) [inline node/node_syncing.go]: why add a function here doing nothing?

File: cmd/client/txgen/main.go [hunk 3]
```diff
@@ -86,141 +96,87 @@ func main() {
...
-			readySignal <- i
-		}
-	}()
+	node := node.New(host, &consensus.Consensus{ShardID: uint32(shardID)}, nil, true) //Make it archival node.
+	node.Client = client.NewClient(node.GetHost(), shardIDs)
+	node.NodeConfig.SetRole(nodeconfig.ClientNode)
+	node.NodeConfig.SetIsBeacon(false)
+	node.ServiceManagerSetup()
+	node.RunServices()
+	go node.GetSync()
```
Comment (LeoHChen) [inline cmd/client/txgen/main.go]: redundant go call here.

File: cmd/client/txgen/main.go [hunk 4]
```diff
@@ -86,141 +96,87 @@ func main() {
...
-
-			lock.Lock()
-			for shardID, txs := range shardIDTxsMap { // Send the txs to corresponding shards
-				go func(shardID uint32, txs types.Transactions) {
-					SendTxsToShard(clientNode, txs)
-				}(shardID, txs)
+		case <-ticker.C:
+			if node.State.String() == "NodeReadyForConsensus" {
+				utils.GetLogInstance().Debug("Generator Will Send Txns.", "node", node.SelfPeer)
+				utils.GetLogInstance().Debug("Node State", "node state", node.State.String())
```
Comment (LeoHChen) [inline cmd/client/txgen/main.go]: merge with the previous message please.

File: cmd/client/txgen/main.go [hunk 5]
```diff
@@ -86,141 +96,87 @@ func main() {
...
+				txs, _ := GenerateSimulatedTransactionsAccount(int(shardID), node, setting)
+				SendTxsToShard(node, txs)
+				go node.GetSync()
```
Comment (LeoHChen) [inline cmd/client/txgen/main.go]: redundant go calls.

File: cmd/client/txgen/main.go [hunk 6]
```diff
@@ -86,141 +96,87 @@ func main() {
...
-	// Start the client server to listen to leader's message
-	go func() {
-		// wait for 3 seconds for client to send ping message to leader
-		// FIXME (leo) the readySignal should be set once we really sent ping message to leader
-		time.Sleep(3 * time.Second) // wait for nodes to be ready
-		for _, i := range shardIDs {
-			readySignal <- i
-		}
-	}()
+	node := node.New(host, &consensus.Consensus{ShardID: uint32(shardID)}, nil, true) //Make it archival node.
```
Comment (LeoHChen) [inline cmd/client/txgen/main.go]: don't use the variable name the same as the package name.  It is very confusing. clientNode is better, IMHO.

**Pull request closed without merging**